diff --git a/linux-2.6.34.7/arch/x86/include/asm/unistd_32.h b/linux-2.6.34.7/arch/x86/include/asm/unistd_32.h
index beb9b5f..d86deab 100644
--- a/linux-2.6.34.7/arch/x86/include/asm/unistd_32.h
+++ b/linux-2.6.34.7/arch/x86/include/asm/unistd_32.h
@@ -343,7 +343,7 @@
 #define __NR_rt_tgsigqueueinfo	335
 #define __NR_perf_event_open	336
 #define __NR_recvmmsg		337
-
+#define __NR_team08     338
 #ifdef __KERNEL__
 
 #define NR_syscalls 338
diff --git a/linux-2.6.34.7/arch/x86/kernel/syscall_table_32.S b/linux-2.6.34.7/arch/x86/kernel/syscall_table_32.S
index 8b37293..adc1b66 100644
--- a/linux-2.6.34.7/arch/x86/kernel/syscall_table_32.S
+++ b/linux-2.6.34.7/arch/x86/kernel/syscall_table_32.S
@@ -337,3 +337,7 @@ ENTRY(sys_call_table)
 	.long sys_rt_tgsigqueueinfo	/* 335 */
 	.long sys_perf_event_open
 	.long sys_recvmmsg
+    .long sys_team08    /* 338 */
+
+
+
diff --git a/linux-2.6.34.7/kernel/sys.c b/linux-2.6.34.7/kernel/sys.c
index 7cb426a..54d345f 100644
--- a/linux-2.6.34.7/kernel/sys.c
+++ b/linux-2.6.34.7/kernel/sys.c
@@ -3,7 +3,6 @@
  *
  *  Copyright (C) 1991, 1992  Linus Torvalds
  */
-
 #include <linux/module.h>
 #include <linux/mm.h>
 #include <linux/utsname.h>
@@ -21,6 +20,7 @@
 #include <linux/capability.h>
 #include <linux/device.h>
 #include <linux/key.h>
+#include <linux/time.h> // Added to get systime
 #include <linux/times.h>
 #include <linux/posix-timers.h>
 #include <linux/security.h>
@@ -43,6 +43,9 @@
 #include <linux/kprobes.h>
 #include <linux/user_namespace.h>
 
+#include <linux/linkage.h>
+#include <asm/thread_info.h>
+
 #include <asm/uaccess.h>
 #include <asm/io.h>
 #include <asm/unistd.h>
@@ -1717,4 +1720,21 @@ int orderly_poweroff(bool force)
 
 	return ret;
 }
+
+/* System call to identify the current process.
+ * Print out the system time.
+ * */
+asmlinkage long sys_team08(void){
+    struct  timespec ts;
+    int hours, minutes, seconds;
+    printk ("sys_team08 called from process %d.\n", current->tgid); 
+    ts = current_kernel_time();
+    printk("time = %ld\n", ts.tv_sec);
+    hours = ((ts.tv_sec / 3600) + 17) % 24;
+    minutes = (ts.tv_sec / 60) % 60;
+    seconds = (ts.tv_sec % 60);
+    printk("time = %d:%0d:%0d\n", hours, minutes, seconds);
+    return 0;
+}
+    
 EXPORT_SYMBOL_GPL(orderly_poweroff);
